File: test/fixtures/small-php/Shape.php
<?php

namespace Sample;

interface Shape
{
    public function area(): float;
}

trait Tagged
{
    public function tag(): string
    {
        return 'shape';
    }
}

class Circle implements Shape
{
    use Tagged;

    public function __construct(private float $radius) {}

    public function area(): float
    {
        return 3.14 * $this->radius * $this->radius;
    }

    public function label(): string
    {
        return 'circle';
    }
}

/** Factory: construction edge to Circle. */
function defaultCircle(): Circle
{
    return new Circle(1.0);
}

File: test/fixtures/small-cs/Shape.cs
namespace Sample;

public interface IShape
{
    double Area();
}

public class Circle : IShape
{
    private readonly double _r;

    public Circle(double r)
    {
        _r = r;
    }

    public double Area()
    {
        return 3.14 * _r * _r;
    }

    public string Describe()
    {
        return "circle";
    }
}

public static class Shapes
{
    public static Circle DefaultCircle()
    {
        return new Circle(1.0);
    }

    public static string Label(this Circle c)
    {
        return c.Describe();
    }
}
